src/perception/lidar_object_detection/lidar_object_detection/lidar_object_detection/lidar_object_detection_node.py

```diff
@@ -0,0 +1,211 @@
+        data_dict = self.demo_dataset.collate_batch([data_dict])
+        load_data_to_gpu(data_dict)
+
+        with torch.no_grad():
+            pred_dicts, _ = self.model.forward(data_dict)
+
+        original_timestamp = msg.header.stamp
+        self.publish_bounding_boxes(msg, pred_dicts, original_timestamp)
+
+    def pointcloud2_to_xyz_array(self, cloud_msg):
+        num_points = cloud_msg.width * cloud_msg.height
+        cloud_array = np.frombuffer(cloud_msg.data, dtype=np.float32)
+        num_fields = cloud_msg.point_step // 4
+        cloud_array = cloud_array.reshape(num_points, num_fields)
+        if cloud_array.shape[1] > 4:
+            cloud_array = cloud_array[:, :4]
+        if cloud_array.shape[1] <= 4:
+            timestamp = np.full((num_points, 1), fill_value=0.0, dtype=np.float32)
+            cloud_array = np.hstack((cloud_array, timestamp))
+        return cloud_array
+
+    def publish_bounding_boxes(self, pointcloud_msg, pred_dicts, original_timestamp):
+        marker_array = MarkerArray()
+        detections = Detection3DArray()
+        detections.header = pointcloud_msg.header
+        for idx, (box, score) in enumerate(zip(pred_dicts[0]["pred_boxes"], pred_dicts[0]["pred_scores"])):
+            if score > 0.6:
+                marker = Marker()
+                marker.header = pointcloud_msg.header
+                marker.header.stamp = original_timestamp
+                marker.id = idx
+                marker.type = Marker.CUBE
+                marker.action = Marker.ADD
+                marker.pose.position.x = float(box[0])
+                marker.pose.position.y = float(box[1])
+                marker.pose.position.z = float(box[2])
+
+                # Calculate orientation quaternion
+                yaw = float(box[6]) + 1e-10
+                marker.pose.orientation.z = np.sin(yaw / 2)
+                marker.pose.orientation.w = np.cos(yaw / 2)
+
+                marker.scale.x = float(box[3])
+                marker.scale.y = float(box[4])
+                marker.scale.z = float(box[5])
+                marker.color.a = 0.8
+                marker.color.r = 1.0
+                marker.color.g = 0.0
+                marker.color.b = float(pred_dicts[0]["pred_labels"][idx]) / 3
+                marker_array.markers.append(marker)
+
+                detection = Detection3D()
+                detection.header = pointcloud_msg.header
+                detection.bbox.center.position.x = float(box[0])
+                detection.bbox.center.position.y = float(box[1])
+                detection.bbox.center.position.z = float(box[2])
+                detection.bbox.size.x = float(box[3])
+                detection.bbox.size.y = float(box[4])
+                detection.bbox.size.z = float(box[5])
+                detected_object = ObjectHypothesisWithPose()
+                detected_object.hypothesis.class_id = str(pred_dicts[0]["pred_labels"][idx])
```

Comment:
these labels should be pulled out of the model and stored centrally @danielrhuynh in some sort of label server that publishes `/vision_msgs/VisionInfo.msg`

Reply:
Not quite sure what this means lol

Reply:
The pred labels?

Reply:
class labels yeah, otherwise we don't know what the class_id actually refers too downstream

Reply:
eg. how do we know `1` refers to a `car`?

Reply:
Gotcha, added